New python 2.7 Script to decode beacon data

NickWaterton opened the pull request:
Just a simple program to decode beacons. Probably open to a lot of improvement.

Commit:
New python 2.7 Script to decode beacon data

stephen-meyerhofer commented:
I wonder if this would be better in its own repository to avoid cluttering this repository with non-example/non-mcScript code.

Isaksson commented:
Or maybe create a good folder structure for this, I use Node-Red and have some custom functions that I could share so we need a good place for that kind of stuff also.
So what you you think? should it be a separate repository or keep at same place with a better folder structure?

NickWaterton commented:
I have no problem with keeping these sort of things separate. Maybe a separate folder for Scripts or something?

Isaksson commented:
Yes, a folder with name External and in that subfolders for example Node-Red, Python, something like that maybe?

Commit:
added library for MQTT, and example of use "TemeraturePublish"

NickWaterton commented:
Added MQTT Library, plus an example of it's use Temperature Publish

NickWaterton commented:
OK, still havent got the hang of these pull requests...

Commit:
Changed structure to put other programs in folder "external"

Commit:
removed beacon_decode folder

Commit:
Deleted .lst file

stephen-meyerhofer commented:
It's cool code, but unless we write concise core library code that fits under the "libraries" folder (and perhaps a corresponding "examples" example) I think your own personal repository is best. Core code deals with the hardware that comes shipped with the mc-Modules, not every solderable device under the sun.

You will be able to update your code at your own pace and not depend on the mc-Things team to merge your codebase.

To solve the issue of visibility to your repository, perhaps a simple "# Community" section to this repository's README with a descriptive link to your repository is a good solution.

NickWaterton commented:
Aaaand this is why I don't post thing on GitHub normally.

I have no idea what the comment by Stephen-Meyerhofer means.

Nothing I have posted so far requires any solderable hardware, and all is relevant to the core hardware.

So far I have beacon_decode (which decodes default beacons). MQTT library, which makes publishing via MQTT and modifying beacon data easier, and TemperaturePubish, which is an example of how to use both of the previous pieces of code.

No hardware in sight, other than what comes with the module, and all easily modifiable to do whatever thing you want.

It does seem to me though, that if someone develops a library for a device (I have one for the BME280), then the libraries folder would be where it goes, so other people can use it/improve it, without having to search all of github, or have 10 versions of the same library. Just like the Arduino model. That works for me.

If you don't have the device, don't include the tab in your program. isn't that what a library is? a collection of stuff, available if you need it?

stephen-meyerhofer commented:
After thinking a little about how nicely the hardware libraries are named (BME280, [TSL2561](https://github.com/luetgendorf/mcScript/tree/master/libraries/light/TSL2561)), I agree that those would be practical to have in this repository.

It's OK to not have much GitHub experience, we are all learning here. I have most of my GitHub experience from Node.js and NPM and observing the underlying GitHub repos. I'll do a pull request example of what I mean for your beacon_decode in the README.

The TemperaturePublish and MQTT files, in my opinion, do not seem to be general enough for library use. Not everyone will be following the same standards of MQTT topics, or publishing Uptime and BattVoltage with Temperature readings.

I'm thinking this repository should be left for libraries that are ubiquitous, no matter how users choose to implement their "business logic." The BME280 and TSL2561 libraries would be great additions.

Commit:
Alpha Version of Accelerometer Library

ghost commented:
I am looking at this Pull Request to bring it back into our repository. Could you break it up more logically? There seems to be a mix of accelerometer support, beacon decoding, and a MQTT example. I can then pull those in or provide comments on the individual projects. To be fair I haven't gone through creating a pull request with specific commits but I assume it is possible?

NickWaterton commented:
Honestly I don't know if you can separate them. My original pull request only had beacon_decode in it.

I added all the other things after to my own repository, but they seem to have been automatically included in the first pull request.

It seems that you can only do a pull request for the whole repository.

I'm no expert with github, though, maybe there are some tricks.

Nick Waterton P.Eng
Sent from my iPhone

On Aug 31, 2016, at 6:52 PM, mc-Abe <notifications@github.com<mailto:notifications@github.com>> wrote:

I am looking at this Pull Request to bring it back into our repository. Could you break it up more logically? There seems to be a mix of accelerometer support, beacon decoding, and a MQTT example. I can then pull those in or provide comments on the individual projects. To be fair I haven't gone through creating a pull request with specific commits but I assume it is possible?

�
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHubhttps://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mcThings_mcScript_pull_3-23issuecomment-2D243926927&d=CwMFaQ&c=IV_clAzoPDE253xZdHuilRgztyh_RiV3wUrLrDQYWSI&r=U_-uBTyQDj4ifd00J7nSYtwgRmrVu_2vacy9QbRAMDQ&m=28FXA2s197XeUn0T2LQ1CJKFgWmAYnxTioYkHz1e8Ws&s=W9hAWNurCgfcWW1L9y3Sj45sgnT-WHfT4IxhaKntJe0&e=, or mute the threadhttps://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ALCUZ0FebsGgqTxCSdUPO7Ncc2n1vajbks5qlgW2gaJpZM4JvpA3&d=CwMFaQ&c=IV_clAzoPDE253xZdHuilRgztyh_RiV3wUrLrDQYWSI&r=U_-uBTyQDj4ifd00J7nSYtwgRmrVu_2vacy9QbRAMDQ&m=28FXA2s197XeUn0T2LQ1CJKFgWmAYnxTioYkHz1e8Ws&s=BaHzLHgVUiHHC-MJiN2UPKsFGk2L60xxDfb6FQVS_R0&e=.

Unsubscribe from our commercial electronic messages.
https://supportcentral.ge.com/esurvey/GE_survey/takeSurvey.html?form_id=18446744073709666319
D�sabonner de nos messages �lectroniques commerciaux.
https://supportcentral.ge.com/esurvey/GE_survey/takeSurvey.html?form_id=18446744073709666263

Commit:
New Library for multi sensor BME280 - temperature, Humidity and Pressure Sensor

Commit:
Added example project using BME280 sensor

Commit:
Added examlple programs for Door nock Sensor and Garage Door Orientaton Sensor

Commit:
Updated MMA8652 Library with various fixes

NickWaterton committed:
Updated MMA8652 Library with various fixes

Fixed Author

NickWaterton committed:
Merge branch 'master' of https://github.com/NickWaterton/mcScript

NickWaterton committed:
Tweaks to Timing

NickWaterton committed:
Updated Door Sensor to tweak timing of knock sensor

NickWaterton committed:
Fixed bug in beacon data reporting

NickWaterton committed:
Fixed another uptime reporting bug

NickWaterton committed:
Fixed Typo's in various files

stephen-meyerhofer commented:
I have another thought when it comes to putting each driver in its own repository, and it deals with semantic versioning. http://semver.org/ is a resource for understanding my thinking.

I have been learning about and working on the MMA8652 driver for the mc-Module110. My approach has been to learn the details of the accelerometer, but save everyone else from having to understand the accelerometer in great detail. After reading helpful code from Nick_W and the mc-Things alpha releases I merged the two approaches into something that I hope is agreeable as a foundation for continual improvement.

The crux of the problem comes down to versioning. Neither my soon to be released driver, Nick_W's driver, or mc-Things alpha driver accomplishes the goal of an easy to use library; Certainly not as simple as the Temperature Module's function "GetTemp". So, as driver's are improved, we want to be able to make breaking changes if necessary. Semantic versioning solves this problem.

Git works well with semantic versioning because commits can be tagged with the semantic versions. Users can then clone specific tags of a repository, so that if they are using a specific version of a driver, they can continue to use the repository even though development has moved forward.

If all drivers are in 1 repository, it becomes nearly impossible to choose which driver versions we would like to use. If each driver gets its own repository, it can be well documented and independently accessible.

My next pull request may add a link to a MMA8652 repository as the first step forward in this process. All will be welcome to contribute.

NickWaterton committed:
Added new Beta for 120 module Accelerometer LIS2DH12 Door Knock Example

NickWaterton committed:
Major rework of most files!

NickWaterton committed:
Major Rework of all files!

NickWaterton committed:
Bug fixes and improvements

NickWaterton committed:
Minor tweaks

NickWaterton committed:
Updated project

NickWaterton committed:
Disabled MQTT queing as seems to be causing random reboots

NickWaterton committed:
reliability improvements

NickWaterton committed:
Added new verions of Battery Monitor and DoorSensor.
Now with json data and time stamps
Added python utilities, beacon_decode.py (nothing new)
and new logging program mcthings_log.py to log and decode
the data from your mqtt modules.

NickWaterton committed:
Added NaN handling

NickWaterton committed:
Added NaN handling, fix typos

NickWaterton committed:
Fixed Error handling

NickWaterton committed:
Fixed stupid errors

NickWaterton committed:
Updated MQTT library, V4 of LIS2DH12 accelerometer library, Demo 205 test program (GPS on motion/stop)

NickWaterton committed:
Added New Time Library, New version of DoorSensor, using Time library

NickWaterton committed:
Removed duplicate beacon_decode.py

NickWaterton committed:
Moved mqtt library to it's own subdirectory

NickWaterton committed:
Updated Demo 205 test program to use New Time library, couple of other minor fixes.

NickWaterton committed:
New lis2dh12 library for module 120. New example programs. new DoorSensor program for module 120
Update to python logging program (better handle time zones)
Lot's of new stuff!

NickWaterton committed:
Minor fixes and some new features, new example programs

NickWaterton committed:
MOre tweaks and example programs

NickWaterton committed:
Added configuration for different destination MQTT broker to source MQTT broker for cloud publishing etc.

NickWaterton committed:
Added client_id to destination broker config

NickWaterton committed:
Added protocol parameter

NickWaterton committed:
Minor improvements to LIS2DH12, new click functions, new example click programs

NickWaterton committed:
Updated lis2dh12 library, Time library, New Examples programs

NickWaterton committed:
New millis() and micros() class (like Arduino)

NickWaterton committed:
Fixed reference to MQTT in Time library
